src/GitObjectDb/Transformations/ChildDeleteTransformation.cs

```diff
@@ -0,0 +1,46 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Text;
+using GitObjectDb.Models;
+using GitObjectDb.Reflection;
+using static GitObjectDb.Transformations.PropertyTransformation;
+
+namespace GitObjectDb.Transformations
+{
+    internal partial class ChildDeleteTransformation : ITransformation
+    {
+        internal ChildDeleteTransformation(IModelObject child)
+            : this(child.GetDataPath(), child.Id)
+        {
+        }
+
+        internal ChildDeleteTransformation(string path, UniqueId childId)
+        {
+            Path = path ?? throw new ArgumentNullException(nameof(path));
+            ChildId = childId;
+        }
+
+        /// <inheritdoc/>
+        public string Path { get; }
+
+        /// <summary>
+        /// Gets the child id to delete.
+        /// </summary>
+        public UniqueId ChildId { get; }
+
+        private static ChildPropertyInfo ExtractProperty(IModelDataAccessor dataAccessor, Expression propertyPicker)
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove the unused private method 'ExtractProperty'. ([csharpsquid:S1144](https://sonarcloud.io/organizations/frblondin-github/rules#rule_key=csharpsquid%3AS1144))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=GitObjectDb&pullRequest=40&issues=AWgkNyhf1_eUtAp8Im7B&open=AWgkNyhf1_eUtAp8Im7B)

[ISSUE_KEY]: AWgkNyhf1_eUtAp8Im7B